### PR TITLE
Move candid ui section to an include file.

### PR DESCRIPTION
**Pull request by lsgunnlsgunn**

### CODE REVIEW

File: modules/developers-guide/examples/candid-ui.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+== Testing functions using a web interface
+
+The canister interface description language—often referred to as Candid or more generally as the IDL—provides a common syntax for specifying input arguments and return values.
+The language performs automatic type mapping so you can use the same syntax to interact with the `+dfx+` command-line interface, JavaScript, {proglang}, and other supported languages or tools.
+
+The language can also automatically generate a default web interface that exposes your program's functions for testing and debugging.
```
Comment (chenyan-dfinity) [inline modules/developers-guide/examples/candid-ui.adoc]:
```suggestion
Based on the type signature of the actor, Candid provides a web interface that allows you to call canister functions in a graphical manner.
```

File: modules/developers-guide/examples/candid-ui.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+To use the default web interface to test canister functions:
```
Comment (chenyan-dfinity) [inline modules/developers-guide/examples/candid-ui.adoc]:
```suggestion
To use the Candid web interface to test canister functions:
```

File: modules/developers-guide/examples/candid-ui.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+. Deploy your project using the `+dfx canister install+` command and copy the canister identifier that begins with the `+ic:+` prefix.
+. Open a browser and navigate to the address and port number specified in the `+dfx.json+` configuration file.
++
+By default, the URL uses the localhost address (`+127.0.0.1+`) and port number `+8000+`, so you can navigate to the front-end for the hello_world program using `+127.0.0.1:8000+` for the URL.
```
Comment (chenyan-dfinity) [inline modules/developers-guide/examples/candid-ui.adoc]:
```suggestion
By default, the URL uses the localhost address (`+127.0.0.1+`) and port number `+8000+`. The URL for Candid UI is located under `/candid` with a required `canisterId` parameter.
```

File: modules/developers-guide/examples/candid-ui.adoc [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+. Append the candid keyword the canister identifier after the host address and port number using the following syntax:
++
+[source,bash,subs="quotes"]
+----
+candid?canisterId=ic:canister_identifier
+----
```
Comment (chenyan-dfinity) [inline modules/developers-guide/examples/candid-ui.adoc]: Remove line 14--19

File: modules/developers-guide/examples/candid-ui.adoc [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
++
+For example, the full URL should look similar to the following but with the `+ic:_canister_identifier_+` that was returned by the `+dfx canister install+` command:
++
+[source,bash,subs="quotes"]
+----
+http://localhost:8000/candid?canisterId=ic:81DDA04F69F40FEEAC
+----
+. Review the list of function calls and types defined in the program.
+. Type a value of the appropriate type for a function, then click *Call* to see the result.
```
Comment (chenyan-dfinity) [inline modules/developers-guide/examples/candid-ui.adoc]:
```suggestion
. Type a value of the appropriate type for a function, then click the *Call* or *Query* button to see the result.
```

File: modules/developers-guide/examples/candid-ui.adoc [hunk 6]
```diff
@@ -0,0 +1,28 @@
+== Testing functions using a web interface
+
+The canister interface description language—often referred to as Candid or more generally as the IDL—provides a common syntax for specifying input arguments and return values.
```
[the remark this answers is not in the document]
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: Changed

File: modules/developers-guide/examples/candid-ui.adoc [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+The language performs automatic type mapping so you can use the same syntax to interact with the `+dfx+` command-line interface, JavaScript, {proglang}, and other supported languages or tools.
```
[the remark this answers is not in the document]
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: Changed.

File: modules/developers-guide/examples/candid-ui.adoc (hunk 1 above)
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: web interface and graphical manner sort of say the same thing, so I restored the reason for using the graphical user interface--testing and debugging.

File: modules/developers-guide/examples/candid-ui.adoc (hunk 2 above)
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: Changed.

File: modules/developers-guide/examples/candid-ui.adoc (hunk 3 above)
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: I would like to keep the two parts of the URL separate and keep the example.

File: modules/developers-guide/examples/candid-ui.adoc (hunk 4 above)
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: I'll remove the format sample

File: modules/developers-guide/examples/candid-ui.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: Replaced where it seems appropriate.

File: modules/developers-guide/examples/candid-ui.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: Changed

File: modules/developers-guide/examples/candid-ui.adoc (hunk 5 above)
Comment (lsgunnlsgunn) [reply modules/developers-guide/examples/candid-ui.adoc]: Changed

File: modules/developers-guide/examples/candid-ui.adoc [hunk 8]
```diff
@@ -0,0 +1,23 @@
+== Testing functions in a browser
+
+The canister interface description language—often referred to as Candid or more generally as the IDL—provides a common language for specifying the signature of a canister service.
+The language provides a unified way of interacting with canisters in various languages and tools, such as the `+dfx+` command-line interface, JavaScript, and {proglang}.
+
+Based on the type signature of the actor, Candid provides a web interface that allows you to call canister functions for testing and debugging.
```
Comment (chenyan-dfinity) [inline modules/developers-guide/examples/candid-ui.adoc]:
```suggestion
Based on the type signature of the actor, Candid provides a web interface that allows you to call canister functions for testing and debugging. Note that the Candid web interface is always available even if you don't setup frontend in your project.
```

File: modules/developers-guide/pages/tutorials/hello-location.adoc [hunk 9]
```diff
@@ -315,6 +315,11 @@ The command displays output similar to the following:
 ----
 ("Hello from San Francisco, Paris, Rome, bon voyage!")
 ----
+
+include::example$candid-ui.adoc[]
+
```
Comment (chenyan-dfinity) [inline modules/developers-guide/pages/tutorials/hello-location.adoc]: The image is not included here?

**Comment by chenyan-dfinity** [comment]
Yes, the docs says `click *Call* or *Query* to see the result`. Depending on your function you will see either query or call.

**Pull request merged**